packages/sdk/src/lib/dataProtectorSharing/smartContract/preflightChecks.ts

```diff
@@ -382,11 +382,17 @@ export const onlyAccountWithMinimumBalance = ({
   accountDetails,
   minimumBalance,
 }: {
-  accountDetails: { balance: bigint };
+  accountDetails: { stackedBalance: bigint; balance: bigint };
   minimumBalance: ethers.BigNumberish;
 }) => {
   const requiredBalance = BigInt(minimumBalance);
-  if (accountDetails.balance < requiredBalance) {
+  if (accountDetails.stackedBalance < requiredBalance) {
+    // The user have xRLC but he didn't stack them
+    if (accountDetails.balance > BigInt(0)) {
+      throw new Error(
+        `No xRLC in your iExec account. You may have xRLC in your wallet but to interact with the iExec protocol, you need to deposit some xRLC into your iExec account.`
+      );
+    }
     throw new Error('Account balance is insufficient.');
```

Comment:
proposed change gives 2 error messages:
- nothing in wallet => `Account balance is insufficient.`
- some xRLC in wallet => `No xRLC in your iExec account. You may have xRLC in your wallet but to interact with the iExec protocol, you need to deposit some xRLC into your iExec account.`

both error messages could be replaced by `Not enough xRLC in your iExec account. You may have xRLC in your wallet but to interact with the iExec protocol, you need to deposit some xRLC into your iExec account.` this would make the native balance detection useless.

Reply:
here is an alternative with less code #319